.buildkite/steps/script.sh

```diff
@@ -9,7 +9,13 @@ upload_tests() {
 }
 
 echo "--- installing dependencies"
-pip install -q --no-cache-dir -r requirements.txt -e .
+pip install -q --no-cache-dir -r requirements.txt
+
+echo "--- listing dependency versions"
+pip freeze
+
+echo "--- installing stellargraph"
+pip install -q --no-cache-dir -e .
```

Comment:
I split this part of the `pip install` to run after the `freeze`, because the `freeze` tries to use `git` to get info about the editable `.` install, and the `python:3.x-slim` images do not contain git. Not having `git` when doing `freeze` on an `.` installation results in a non-problematic (but disconcerting) pair of ERROR/WARNING lines: https://buildkite.com/stellar/stellargraph-public/builds/314#a98235b9-2092-4852-98f4-19baf1a0c452/108-110

By running the `freeze` first, that command only lists the actual dependencies, which is fine.

Reply:
Does it need to be an editable install in the first place? We also have dependencies listed in `setup.py` so I think technically the `.` installation could install additional dependencies

Reply:
> Does it need to be an editable install in the first place? 

Ah, no it does not need to be, and `pip freeze` doesn't use `git` for a non-editable install. 👍 

> We also have dependencies listed in setup.py so I think technically the . installation could install additional dependencies

That's true, although I think it's theoretically a mistake in `requirements.txt` to have a difference here.

Reply:
Since we've got "REQUIRES = [..." in setup.py, I'm not sure this does what's intended?  Would we be better off with an image including git?